web/src/main/java/org/springframework/security/web/DefaultRedirectStrategy.java

```diff
@@ -56,7 +62,13 @@ public void sendRedirect(HttpServletRequest request, HttpServletResponse respons
 		if (this.logger.isDebugEnabled()) {
 			this.logger.debug(LogMessage.format("Redirecting to %s", redirectUrl));
 		}
-		response.sendRedirect(redirectUrl);
+		if (this.statusCode == null) {
```

Comment:
Please change this to check if `this.statusCode == HttpStatus.TEMPORARY_REDIRECT`. This reason is so that if `setStatusCode(302)` is called, the default behavior is restored.

Reply:
Do you actually mean `this.statusCode == HttpStatus.FOUND` and `setStatusCode(HttpStatus.FOUND)` respectively?